docs/code-examples/repository-pattern/01_simple_example/ilGeoLocationRepository.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class ilGeoLocationRepository {
+
+    public const $table = 'geo_location';
+
+    public function __construct(ilDB $a_db)
+    {
+        $this->db = $a_db;
+    }
+
+    public function createGeoLocation(ilObjGeoLocation $obj)
+    {
+        $this->db->insert($this->table, array(
+            'id' => array('integer', $obj->getId()),
+            'title' => array('text', $obj->getTitle()),
+            'lattitude' => array('float', $obj->getLattitude()),
+            'longitude' => array('float', $obj->getLongitude()),
+            'expiration_timestamp' => array('timestamp', $obj->getIExpirationTimestamp())
+        ));
+    }
+
+    public function getGeoLocationById($a_id)
+    {
+        $query = "Select * FROM $this->table WHERE id = " . $this->db->quote($a_id, 'integer');
+        $result = $this->db->query($query);
+
+        if($row = $this->db->fetchAssoc($result))
+        {
+            return new ilObjGeoLocation($row['id'], $row['title'], $row['lattitude'], $row['longitude'], $row['expiration_timestamp'])
+        }
+        else
+        {
+            return NULL;
+        }
+    }
+
+    public function getGeoLocationsByCoordinates($a_lattitude, $a_longitude)
+    {
+        $query = "Select * FROM $this->table WHERE lattitude = " . $this->db->quote($a_lattitude, 'float')
+                    . ' AND longitude = ' . $this->db->quote($a_longitude, 'float');
+        $result = $this->db->query($query);
+
+        $locations = array();
+        while($row = $this->db->fetchAssoc($result))
+        {
+            $locations[] = new ilObjGeoLocation($row['id'], $row['title'], $row['lattitude'], $row['longitude'], $row['expiration_timestamp'])
+        }
+
+        return $locations;
+    }
+
+    public function checkIfLocationExistsById($a_id) : bool
+    {
+        $query = "Select count(*) AS count FROM $this->table WHERE id = " . $this->db->quote($a_id, 'integer');
+        $result = $this->db->query($query);
+
+        return $result['count'] > 0;
+    }
+
+    /**
+     * The name extendGeoLocationExpirationTimestamp would be more beautiful but doesn't match the notation
+     */
+    public function updateGeoLocationExpirationTimestamp($a_id, $a_new_expiration_timestamp)

```

Comment:
I currently doubt if I like that pattern. I would prefer to have change the timestamp on the object itself and then pass the complete object to the repository for an update. Currently this is only a vague feeling, though, and I don't have any good arguments. What do you think?

Reply:
If it is about reading only one object, I totally agree with you. But if the update targets multiple objects like "change every timestamp on objects in specific location", this would mean to create an object out of every data set just to update it. I know this might be an edge case. But it can also be important for performance in those scenarios.

Reply:
Ok, that sounds very plausible.